service/src/main/java/uk/gov/hmcts/reform/fpl/service/document/ManageDocumentService.java

```diff
@@ -238,6 +244,270 @@ public boolean allowMarkDocumentConfidential(CaseData caseData) {
+                AdditionalApplicationsBundle.class));
+        }
+    }
+
+    private static AdditionalApplicationsBundle applyNewList(String propertyName,
+                                                             AdditionalApplicationsBundle aab,
+                                                             C2DocumentBundle c2DocumentBundle,
+                                                             List<Element<SupportingEvidenceBundle>> newList) {
+        try {
+            Method builderPropertySetter = AdditionalApplicationsBundle.AdditionalApplicationsBundleBuilder.class
+                .getMethod(propertyName, C2DocumentBundle.class);
+            C2DocumentBundle newC2DocumentBundle = c2DocumentBundle.toBuilder()
+                .supportingEvidenceBundle(newList)
+                .build();
+            return ((AdditionalApplicationsBundle.AdditionalApplicationsBundleBuilder)
+                builderPropertySetter.invoke(aab.toBuilder(), newC2DocumentBundle)).build();
+        } catch (NoSuchMethodException | InvocationTargetException | IllegalAccessException e) {
+            throw new AssertionError(format("Fail to get property %s from %s", propertyName,
+                AdditionalApplicationsBundle.AdditionalApplicationsBundleBuilder.class));
+        }
+    }
+
+    private void appendToRemovedList(DocumentType documentType,
+                                     CaseData caseData,
+                                     Element removed,
+                                     Map<String, Object> output) {
+        List<Element<?>> removedList = getListOfRemovedElement(caseData, documentType);
+        removedList.add(removed);
+        output.put(documentType.getJsonFieldNameOfRemovedList(), removedList);
+    }
+
+    private static Element<ManagedDocument> toManagedDocumentElement(Element<SupportingEvidenceBundle> removed) {
+        SupportingEvidenceBundle seb = removed.getValue();
+        return element(removed.getId(), ManagedDocument.builder()
+            .uploaderType(seb.getUploaderType())
+            .uploaderCaseRoles(seb.getUploaderCaseRoles())
+            .translationRequirements(seb.getTranslationRequirements())
+            .markAsConfidential(seb.getMarkAsConfidential())
+            .document(seb.getDocument())
+            .build());
+    }
+
+    private List<String> getC2DocumentBundleProperties() {
+        List<String> allProperties = new ArrayList<>(List.of("c2DocumentBundle", "c2DocumentBundleConfidential",
+            "c2DocumentBundleLA"));
+        for (int i = 0; i <= 14; i++) {
+            allProperties.add("c2DocumentBundleChild" + i);
+        }
+        for (int i = 0; i <= 9; i++) {
+            allProperties.add("c2DocumentBundleResp" + i);
+        }
+        return allProperties;
+    }
+
+    @SuppressWarnings("unchecked")
+    private Element<? extends WithDocument> handleC1OrC2SupportingDocumentsInAdditionalApplications(
```

Comment:
`c2ApplicationDocList` (of course the `c2ApplicationDocListLA` and ``c2ApplicationDocListCTSC` as well) is used when user uploads additional application supporting documents via manage document event. 

those supporting documents under c2document are uploaded by "Upload Additional Application" event which will not be synchronised to `c2ApplicationDocList`.

Therefore, the remove document logic will consider both documents from `c2ApplicationDocList` and those under the series of `c2DocumentBundle` and `otherApplicationBundle`.

By the way, supplements are unable to be removed which is confirmed by Sara.

Reply:
Think you tested it, so it should be fine
But, my eyes are bad, I can't see where is the logic removing `c2ApplicationDocList`.

In `handleRemoveGeneralDocumentType`, it will invoke `handleC1OrC2SupportingDocumentsInAdditionalApplications` directly then return. It won't go through those generic field resolving.

And in `handleC1OrC2SupportingDocumentsInAdditionalApplications`, it only handle the list of `AdditionalApplicationsBundle` but not `c2ApplicationDocList`.

Reply:
For removing `c2ApplicationDocList`, the fieldName is `c2ApplicationDocList` instead of `C2_APPLICATION_DOCUMENTS.name()`. The enum names were used if the documents are under additional application bundles.